### ISSUE TITLE
Problems when trying to install

### ISSUE THREAD

**Issue by nicwats** [markgdev/home-assistant_OctopusAgile#100]
**Describe the bug**
When trying add the integration, a popup appears which has 3x text entry boxes, the bottom one of which looks like it's for a password as it has a 'reveal' icon next to it, however there's nothing stating what should be in each of the boxes.  

Running on a Raspberry Pi 3 with all latest updates installed. Trying to install using Chrome & Safari on a Mac and also on ipad.

**Comment by nicwats** [comment]
![Uploading 99868598-636E-49F5-B903-CF0522A571BC.png…]()

**Comment by nicwats** [comment]
![Uploading 405B4B6D-EEC5-4025-A0C2-9B2CE200DF21.png…]()

**Comment by pgm83** [comment]
I'm installing right now too, although I note that it says the GUI configuration flow is currently not implemented, and you need to manually add the configuration yaml code. I've done that but now got errors to try to figure out!

**Comment by nicwats** [comment]
When you hit install, did the three text boxes have a title? Any chance you remember what they were as I can just enter it and move on if I can find out what's supposed to go in there.

**Comment by pgm83** [comment]
The way I read it, they don't do anything but I may be wrong. I put a space in each to get past that issue and then manually added the code to the right places in configuration.yaml, and entered the API key etc while doing so. 

Prob worth noting that the other issue listed here suggest that this is custom component is broken/doesn't work with the latest home assistant versions, and needs fixing . The main author no longer supports this and has tagged help wanted on the issues to see if someone else will fix it!

**Comment by LucidityCrash** [comment]
> Prob worth noting that the other issue listed here suggest that this is custom component is broken/doesn't work with the latest home assistant versions, and needs fixing . The main author no longer supports this and has tagged help wanted on the issues to see if someone else will fix it!


Gotta say its still working just fine for me ... but I already had it working before the reports of it not working started.  I wonder if it is related to the octopusagile.json file not being there initially.

**Comment by blardybla** [comment]
Can anyone advise me on which configuration.yaml file we need to update with the Octopus data is? Is it in the \config file of the main HA? Or should I be seeing something in \config\custom_components\octopusagile ?

**Comment by Webreaper** [comment]
Having the same issue - did anyone figure this out for the latest HA?

**Comment by PhillyGilly** [comment]
@Webreaper I just added this integration and it seems to be working. First step was to manually add the custom-card (as per instructions) and then use file editor to write the configuration files. My preference is not to add anything else into configuration.yaml apart from an include:
```octopusagile: !include octopusagile.yaml```
and then to put the configuration in octopusagile.yaml (note there are no indents required)
```
region_code: "B"
mpan: aaaaaa
serial: bbbbbb
auth: cccccc
startdate: "2024-01-01"
sensor:
- platform: "octopusagile"
```
Then I installed the integration from HACS. After downloading and restarting HA, I went to settings-integrations and added the Octopus Agile integration which produced the empty, three line dialogue box.  I defeated that by putting a blank space on each line. This resulted in simultaneous success and failure:
![image](https://github.com/markgdev/home-assistant_OctopusAgile/assets/56273663/dcb37db5-7c9b-4249-8475-5c968d661714)
After OK-ing success, I then used the ... to delete the failed entry, and when I returned to the Integrations page Octopus Agile had disappeared. As the GitHub says you can't configure it via the GUI, I wasn't overly concerned and a bit later this is what I can see:
![image](https://github.com/markgdev/home-assistant_OctopusAgile/assets/56273663/ee7d42fc-252c-4173-a935-a24c1f54a465)
And the octopusagile sensor has some values, hopefully there will be more after 16:30:
![image](https://github.com/markgdev/home-assistant_OctopusAgile/assets/56273663/7b210c16-e42b-44b7-a6b4-e2eaba215336)

**Comment by PhillyGilly** [comment]
Looking good here:
![image](https://github.com/markgdev/home-assistant_OctopusAgile/assets/56273663/eb62be01-8de7-49dc-afb6-ccb8e016716f)

**Comment by Webreaper** [comment]
I got it working using BottleDave's maintained integration, without needing to use this one. Works a treat. 

![Screenshot_20240123_171800_Home Assistant.jpg](https://github.com/markgdev/home-assistant_OctopusAgile/assets/4901102/a539b11e-0676-4886-9ebd-fae5d7ac197e)

**Comment by PhillyGilly** [comment]
That's great. I was travelling the opposite way!I've got BottlecapDave's integration already bringing in my live data from Octopus which is E7 and I wanted to compare that with Agile. Hence my interest. I've been playing with PredBat too, which gets the data but keeps interfering with my battery charging automation. I think that my best option will be to get that Agile prices by a rest command, and then pick out the ten cheapest overnight slots to fully charge my batteries. Cheers!

**Comment by SingingDwarf** [comment]
@PhillyGilly I followed your instructions, but can't add the custom card.

When trying to add `type: "custom:agile-rates-card"` I get the error

`Custom element doesn't exist: agile-rates-card.
type: custom:agile-rates-card`

**Comment by PhillyGilly** [comment]
I assume you did this: "First step was to manually add the custom-card (as per instructions) and then use file editor to write the configuration files"? I just looked for  the custom card in HACs and couldn't see it. Have you tried this instead: https://github.com/lozzd/octopus-energy-rates-card

**Comment by SingingDwarf** [comment]
Yes, I followed the process here https://github.com/markgdev/home-assistant_OctopusAgile/tree/master/custom_cards
But I didn't see the custom card visible when trying to add to the dashboard.

I tried the lozzd card, but as I'm using the https://github.com/markgdev/home-assistant_OctopusAgile integration, the octopusagile.rates entity contains current and future pricing, so doesn't seem to work?
